### PR TITLE
Replace dpiutil scale calls by scale with zoom calls

### PR DESCRIPTION
**Pull request by amartya4256**
Depends on #1339

### CODE REVIEW

File: bundles/org.eclipse.swt/Eclipse SWT/win32/org/eclipse/swt/widgets/Tree.java [hunk 1]
```diff
@@ -5371,7 +5371,7 @@ public void setTopItem (TreeItem item) {
  * In a Tree without imageList, the indent also controls the chevron (glyph) size.
  */
 private void calculateAndApplyIndentSize() {
-	int indent = DPIUtil.autoScaleUpUsingNativeDPI(DEFAULT_INDENT);
+	int indent = DPIUtil.scaleUp(DEFAULT_INDENT, nativeZoom);
```
Comment (HeikoKlare) [inline bundles/org.eclipse.swt/Eclipse SWT/win32/org/eclipse/swt/widgets/Tree.java:5374]: Isn't this "conflicting" with #1330? Maybe we should have a separate PR for the replacement of the `DPIUtil.autoScaleUpUsingNativeDPI()` calls, in particular because I think that method is completely unused after the changes in this PR (and makes no sense anymore anyway), so it should probably be removed?

File: bundles/org.eclipse.swt/Eclipse SWT AWT/win32/org/eclipse/swt/awt/SWT_AWT.java [hunk 2]
```diff
@@ -250,7 +250,7 @@ public static Frame new_Frame (final Composite parent) {
 
 	parent.getDisplay().asyncExec(() -> {
 		if (parent.isDisposed()) return;
-		final Rectangle clientArea = DPIUtil.autoScaleUp(parent.getClientArea()); // To Pixels
+		final Rectangle clientArea = DPIUtil.scaleUp(parent.getClientArea(), parent.nativeZoom); // To Pixels
```
Comment (HeikoKlare) [inline bundles/org.eclipse.swt/Eclipse SWT AWT/win32/org/eclipse/swt/awt/SWT_AWT.java:253]: Why use native zoom instead of the "adapted one" from `DPIUtil.getZoomForAutoscaleProperty()` here?

File: bundles/org.eclipse.swt/Eclipse SWT/win32/org/eclipse/swt/widgets/Widget.java [hunk 3]
```diff
@@ -64,7 +64,7 @@ public abstract class Widget {
 	 *
 	 * @noreference This field is not intended to be referenced by clients.
 	 */
-	protected int nativeZoom;
+	public int nativeZoom;
```
Comment (HeikoKlare) [inline bundles/org.eclipse.swt/Eclipse SWT/win32/org/eclipse/swt/widgets/Widget.java:67]: Why is this necessary?

File: bundles/org.eclipse.swt/Eclipse SWT/win32/org/eclipse/swt/widgets/Tree.java (hunk 1 above)
Comment (amartya4256) [reply bundles/org.eclipse.swt/Eclipse SWT/win32/org/eclipse/swt/widgets/Tree.java:5374]: I am not sure if it is used for some other platform though. I can have a look.

File: bundles/org.eclipse.swt/Eclipse SWT AWT/win32/org/eclipse/swt/awt/SWT_AWT.java (hunk 2 above)
Comment (amartya4256) [reply bundles/org.eclipse.swt/Eclipse SWT AWT/win32/org/eclipse/swt/awt/SWT_AWT.java:253]: Oh, a mistake

File: bundles/org.eclipse.swt/Eclipse SWT/win32/org/eclipse/swt/widgets/Widget.java (hunk 3 above)
Comment (amartya4256) [reply bundles/org.eclipse.swt/Eclipse SWT/win32/org/eclipse/swt/widgets/Widget.java:67]: Because we need the nativeZoom to be used somewhere outside the package.

File: bundles/org.eclipse.swt/Eclipse SWT/win32/org/eclipse/swt/widgets/Tree.java (hunk 1 above)
Comment (amartya4256) [reply bundles/org.eclipse.swt/Eclipse SWT/win32/org/eclipse/swt/widgets/Tree.java:5374]: Alright its just used for win32

File: bundles/org.eclipse.swt/Eclipse SWT/win32/org/eclipse/swt/widgets/Widget.java (hunk 3 above)
Comment (HeikoKlare) [reply bundles/org.eclipse.swt/Eclipse SWT/win32/org/eclipse/swt/widgets/Widget.java:67]: "somwhere"?

Comment (akoch-yatta) [reply bundles/org.eclipse.swt/Eclipse SWT/win32/org/eclipse/swt/widgets/Widget.java:67]: just saw in SWT_AWT as well

Comment (HeikoKlare) [reply bundles/org.eclipse.swt/Eclipse SWT/win32/org/eclipse/swt/widgets/Widget.java:67]: I see. Thank you for the hints! Then it seems to be unavoidable to make it public.

**Review by HeikoKlare — APPROVED** [review]
Looks good. All adaptations are sound.

File: bundles/org.eclipse.swt/Eclipse SWT Drag and Drop/win32/org/eclipse/swt/dnd/DropTarget.java [hunk 4]
```diff
@@ -281,8 +281,9 @@ int DragEnter_64(long pDataObject, int grfKeyState, long pt, long pdwEffect) {
 }
 
 int DragEnter(long pDataObject, int grfKeyState, int pt_x, int pt_y, long pdwEffect) {
-	pt_x = DPIUtil.autoScaleDown(pt_x);// To Points
-	pt_y = DPIUtil.autoScaleDown(pt_y);// To Points
+	int zoom = DPIUtil.getZoomForAutoscaleProperty(nativeZoom);
+	pt_x = DPIUtil.scaleDown(pt_x ,zoom);// To Points
```
[the remark this answers is not in the document]
Comment (amartya4256) [reply bundles/org.eclipse.swt/Eclipse SWT Drag and Drop/win32/org/eclipse/swt/dnd/DropTarget.java:285]: Done

File: bundles/org.eclipse.swt/Eclipse SWT/win32/org/eclipse/swt/graphics/Image.java [hunk 5]
```diff
@@ -204,8 +204,8 @@ public final class Image extends Resource implements Drawable {
 public Image(Device device, int width, int height) {
 	super(device);
 	initialNativeZoom = DPIUtil.getNativeDeviceZoom();
-	width = DPIUtil.autoScaleUp (width);
-	height = DPIUtil.autoScaleUp (height);
+	width = DPIUtil.scaleUp (width, getZoom());
```
[the remark this answers is not in the document]
Comment (amartya4256) [reply bundles/org.eclipse.swt/Eclipse SWT/win32/org/eclipse/swt/graphics/Image.java:207]: Done
